src/controller/c/tcp_client.h

```diff
@@ -27,18 +27,14 @@
 extern "C" {
 #endif
 
-struct _TcpClient {
-  int fd;
-};
-
-typedef struct _TcpClient TcpClient;
-
-TcpClient *tcp_client_new(const char *host, int port);
-int tcp_client_open();
-int tcp_client_connect(TcpClient *c, const char *host, int port);
-bool tcp_client_send(TcpClient *c, const char *buffer, int size);
-int tcp_client_receive(TcpClient *c, char *buffer, int size);
-void tcp_client_close(TcpClient *c);
+typedef int TcpClient;
```

Comment:
I believe it makes the code less opaque to remove this `typedef` and use `int` instead for the file descriptor of the socket, which is something very standard.
```suggestion
```